types/wlr_surface.c

```diff
@@ -67,6 +67,9 @@ static void surface_set_opaque_region(struct wl_client *client,
 		struct wl_resource *resource,
 		struct wl_resource *region_resource) {
 	struct wlr_surface *surface = wl_resource_get_user_data(resource);
+	if ((surface->pending.invalid & WLR_SURFACE_INVALID_OPAQUE_REGION)) {
+		pixman_region32_clear(&surface->pending.opaque);
```

Comment:
It's what we do below anyway (we just copy over, while the other two damage callback do unions)
The clear only frees memory if that is called twice. I'll check what the protocol wants us to do.